src/ORM/Hierarchy/MarkedSet.php

```diff
@@ -457,6 +458,13 @@ public function markPartialTree()
         $this->clearMarks();
         $this->markUnexpanded($rootNode);
 
+        if ($this->getNumChildrenMethod() === 'numChildren' && is_a($rootNode, DataObject::class)) {
+            $baseClass = Hierarchy::getBaseHierarchyClass($rootNode);
+            if (!Hierarchy::numChildrenCacheIsPopulated($baseClass)) {
+                Hierarchy::prepopulate_numchildren_cache($baseClass);
+            }
+        }
```

Comment:
In CMSMain this gets cached already because `CMSMain::TreeAsUL()` calls `prepopulateTreeDataCache()`.

Can you please give me context of:
1. When are we expecting this to be called without the above caching already having happened?
2. Why are we calling `prepopulate_numchildren_cache()` instead of `prepopulateTreeDataCache()`?

Reply:
1. `CMSMain::getsubtree()` gets hit when doing AJAX requests which opening child nodes on site tree which will also calls `MarkedSet::markPartialTree()`
2. Have updated to use `prepopulateTreeDataCache()`